src/test/groovy/edu/ie3/simona/model/participant/PvModelIT.groovy

```diff
@@ -174,6 +172,7 @@ trait PvModelITHelper {
       HashMap modelToWeatherMap
       if (weatherMap.containsKey(time)) {
         modelToWeatherMap = weatherMap.get(time)
+        weatherMap.put(time, modelToWeatherMap)
```

Comment:
Why is this needed? Seems like we're adding something back to the map, that we've retrieved from the map beforehand.